app/controllers/textbook_transactions_controller.rb

```diff
@@ -0,0 +1,122 @@
+class TextbookTransactionsController < ApplicationController
+
+  def index
+    @textbook_transactions = TextbookTransaction.active.map do |tt|
+      {
+        :id => tt.id,
+        :price => "$" + tt.price.to_s,
+        :courses => tt.book.sections.map(&:course).uniq.map(&:mnemonic_number).join(", "),
+        :title => tt.book.title,
+        :book_id => tt.book_id,
+        :author => tt.book.author,
+        :condition => tt.condition,
+        :notes => tt.notes,
+        :end_date => (tt.created_at + 3.days).localtime.strftime("%b %d, %I:%M %p")
+      }
+    end
+  end
+
+  def books
+    @books = Book.order("RAND()").map do |book|
+      {
+        :image => (book.small_image_link ? book.small_image_link : "/assets/icons/no_book.png"),
+        :title => book.title,
+        :id => book.id
+      }
+    end
+  end
+
+  def get_book_titles
+    query = params[:query]
+
+    results = Book.order("RAND()").map do |book|
+      {
+        :id => book.id,
+        :title => book.title,
+        :image => (book.small_image_link ? book.small_image_link : "/assets/icons/no_book.png")
+      }
+    end
+
+    render :json => [results]
+  end
+
+  def claim
+    puts params
+    cell = params[:cellphone].strip
+    current_user.update(:cellphone => cell)
+    transaction = TextbookTransaction.active.find(params[:claim_id])
+
+    if cell.length == 10
+      response = RestClient.post 'http://textbelt.com/text', :number => transaction.seller.cellphone, :message => "Your posting for \"#{transaction.book.title}\" has been claimed!\nContact info: #{current_user.cellphone}"
+      if JSON.parse(response)["success"]
+        puts 'hihihihihihihihihihi'
+        transaction.update(:buyer_id => current_user.id)
+        transaction.update(:sold_at => Time.now)
+        render :json => {
+          status: "success"
+        }
+      else 
+        render :json => {
+          status: "Internal error"
+        }
+      end
+    else
+      render :json => {
+        status: "Badly formatted phone number"
+      }
+    end
+  end
+
+  def create
+    current_user.update(:cellphone => params[:cellphone])
+
+    if params[:cellphone].length == 10
```

Comment:
you should move these kind of validations to javascript, to increase responsiveness